sources/dicom/large_image_source_dicom/assetstore/dicomweb_assetstore_adapter.py

```diff
@@ -0,0 +1,220 @@
+from requests.exceptions import HTTPError
+
+from girder.exceptions import ValidationException
+from girder.models.file import File
+from girder.models.folder import Folder
+from girder.models.item import Item
+from girder.utility.abstract_assetstore_adapter import AbstractAssetstoreAdapter
+
+DICOMWEB_META_KEY = 'dicomweb_meta'
+
+
+class DICOMwebAssetstoreAdapter(AbstractAssetstoreAdapter):
+    """
+    This defines the interface to be used by all assetstore adapters.
+    """
+
+    def __init__(self, assetstore):
+        super().__init__(assetstore)
+
+    @staticmethod
+    def validateInfo(doc):
+        # Ensure that the assetstore is marked read-only
+        doc['readOnly'] = True
+
+        required_fields = [
+            'url',
+        ]
+
+        info = doc.get(DICOMWEB_META_KEY, {})
+
+        for field in required_fields:
+            if field not in info:
+                raise ValidationException('Missing field: ' + field)
+
+        # If these are empty, they need to be converted to None
+        convert_empty_fields_to_none = [
+            'qido_prefix',
+            'wado_prefix',
+            'auth_type',
+        ]
+
+        for field in convert_empty_fields_to_none:
+            if isinstance(info.get(field), str) and not info[field].strip():
+                info[field] = None
+
+        # Now, if there is no authentication, verify that we can connect to the server.
+        # If there is authentication, we may need to prompt the user for their
+        # username and password sometime before here.
+        if info['auth_type'] is None:
+            study_instance_uid_tag = '0020000D'
```

Comment:
We have pydicom as a soft dependency anyway.  Should we use `study_instance_uid_tag = f'{pydicom.datadict.tag_for_keyword("StudyID"):08x}'`

Reply:
Fixed!